app/models/grda_warehouse/hud/client.rb

```diff
@@ -1778,17 +1778,23 @@ def total_days_of_service
       ((date_of_last_service - date_of_first_service).to_i + 1) rescue 'unknown' # rubocop:disable Style/RescueModifier
     end
 
-    def self.text_search(text, client_scope:)
-      text_searcher(text) do |where|
-        client_scope.
-          joins(:warehouse_client_source).searchable.
-          where(where).
-          preload(:destination_client).
-          map { |m| m.destination_client&.id }.
-          compact
-      rescue RangeError
-        return none
-      end
+    # @param client_scope [GrdaWarehouse::Hud::Client.source] source clients to search in
+    # @param sorted [Boolean] order results by closest match to text
+    def self.text_search(text, client_scope: nil, sorted: true)
+      # Get search results from client scope. Then return the unique destination client records that map to those matching source records
+      results = (client_scope || self).searchable.text_searcher(text, sorted: sorted, resolve_for_join_query: true)
+      grouped = GrdaWarehouse::WarehouseClient
+        .joins(%(JOIN (#{results.to_sql}) src_search_results ON "warehouse_clients"."source_id" = "src_search_results"."client_id"))
+        .select(Arel.sql(%("warehouse_clients"."destination_id" AS client_id, MAX(src_search_results.score) AS score)))
+        .group(Arel.sql('1'))
+
+      mapped = joins(%(JOIN (#{grouped.to_sql}) AS dst_search_results ON dst_search_results.client_id = "Client".id))
+      mapped = mapped.order(Arel.sql('dst_search_results.score DESC'), :id) if sorted
+      mapped
```

Comment:
This looks correct to me. I tested out searching by warehouse destination id and source client ID in the Warehouse and HMIS, both work as expected.